### PR TITLE
Support gzip

### LINKED ISSUE #3: Add gzip content encoding
Can we add an option to support `Content-Encoding` with `gzip` support?

This shouldn't require a change to Wagi. But it might require a fairly major rewrite of Bartholomew. Instead of printing directly to STDOUT, we'd have to print to an output stream that is wrapping a gzip writer. Then we would have to flush the gzip writer to stdout.

We'd also have to write the extra `Content-Encoding: gzip` header.

### PR DESCRIPTION
**Pull request by rajatjindal**
fixes #3

### CODE REVIEW

**Comment by rajatjindal** [comment]
Requesting review: @technosophos

File: src/main.rs [hunk 1]
```diff
@@ -113,8 +120,12 @@ fn exec() -> anyhow::Result<()> {
                     if content_type.starts_with("text/html") {
                         data = minify::html::minify(&data);
                     }
-                    send_result(path_info, data, content_type, status_opt);
-
+                    
+                    if gzip_encoding {
```
Comment (technosophos) [inline src/main.rs:138]: If I am reading the docs right (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding), I think Bartholomew also must check the `HTTP_ACCEPT_ENCODING` header to see if the client supports `gzip`.

Comment (technosophos) [reply src/main.rs:138]: Sorry... I meant the environment variable. The header Accept-Encoding becomes the env var HTTP_ACCEPT_ENCODING.

**Review by technosophos — COMMENTED** [review]
This is great! I think that in order to see if a client supports gzip-compressed data, we have to check the accept-encoding header.

File: src/main.rs (hunk 1 above)
Comment (rajatjindal) [reply src/main.rs:138]: thanks. I am now checking for the accept-encoding header before enabling gzip response.

**Comment by rajatjindal** [comment]
Hi @technosophos 

is this looking ok or we need to do more changes? 

if you have an alternate approach in mind, i am happy to try that out

**Comment by technosophos** [comment]
This looks great! I am going to manually test it, then approve.

**Comment by technosophos** [comment]
Manual testing went really well! Merging.

**Pull request merged**